pkg/validation/limits.go

```diff
@@ -122,6 +123,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 
 	f.IntVar(&l.RulerMaxRulesPerRuleGroup, "ruler.max-rules-per-rule-group", 0, "Maximum number of rules per rule group per-tenant. 0 to disable.")
 	f.IntVar(&l.RulerMaxRuleGroupsPerTenant, "ruler.max-rule-groups-per-tenant", 0, "Maximum number of rule groups per-tenant. 0 to disable.")
+	f.IntVar(&l.RulerRemoteWriteQueueCapacity, "ruler.remote-write.queue-capacity", 10000, "Capacity of remote-write queues; if a queue exceeds its capacity it will evict oldest samples.")
```

Comment:
I'm a bit hesitant to add this to the overrides because the overrides are expected to be updated _during runtime_. This would mean that all internal queue capacities would need to be adjusted at some regular interval (for instance if the queue capacity configuration was increased or decreased). I'm not opposed to either approach:
1) Removing this as a runtime configuration and making it a static config across all tenants in the ruler config block
2) Exposing it as a runtime config per tenant and re-adjusting queue capacities on the fly

(2) Would entail some more work, though.